### PR TITLE
NiFI-1025: Imported aws-sdk:1.10.27 and joda-time: 2.8.2 

### PR DESCRIPTION
**Pull request by vs186031**
NiFI-1025: Imported aws-sdk:1.10.27 and joda-time: 2.8.2 to resolve AWS issues with Java 1.8 update 60.

Re-did the tests to make them : 
a_ clean up after themselves.
b_ fail if initial setup conditions are not met.

### CODE REVIEW

File: nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/test/java/org/apache/nifi/processors/aws/s3/TestPutS3Object.java [hunk 1]
```diff
@@ -87,16 +93,29 @@ public void testStorageClass() throws IOException {
...
-}
+    @Test
+    /**
+     * Run this test to check what version of Joda-time is actually being linked.
+     *
+     * @see: https://issues.apache.org/jira/browse/NIFI-1025
+     * @see: https://github.com/aws/aws-sdk-java/issues/444
+     */
+    public void testJodaTime() {
+        System.out.println(new DateTime().getClass().getProtectionDomain().getCodeSource());
```
Comment (trkurc) [inline nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/test/java/org/apache/nifi/processors/aws/s3/TestPutS3Object.java]: This doesn't appear to test for something - just a dump to the screen

**Comment by trkurc** [comment]
I'm looking through the code. I'm wondering if you considered trying to convert these unit tests which require network calls to use mock services. It dawned on me as I was checking this through that these changes are all in @Ignore's

**Comment by vs186031** [comment]
@trkurc : Indeed they are all actual Network calls and have been annotated with *@Ignore*. I suspect ideally we want both Mock Services and _real_ network calls, with the real services being more important.

**Comment by trkurc** [comment]
A couple of comments:

I recommend changing S3Setup to something like AbstractS3Test and making it an abstract class (if for nothing else, because it doesn't contain test code and is the base class for the other tests)

What this method did was a bit misleading. I expected that the parameter was the filename. I'd rather see it with two parameters, key and filename, to make it a bit more general purpose. If these tests are extended with multiple files, having that second parameter would be helpful. Also, should it throw an AmazonS3Exception?

```
    protected void uploadTestFile(String key) throws IOException  {
```

In the following code, rather than printing to System.err, would Asserting make more sense? Or did you intend for the doesBucketExist be a catchall for these problems?
```
        } catch (final AmazonS3Exception e) {
            System.err.println("Can't create the key " + BUCKET_NAME + ":" + e.toString());
        } catch (final IOException e) {
            System.err.println(CREDENTIALS_FILE + " doesn't exist.");
        }

        if (!client.doesBucketExist(BUCKET_NAME)) {
            Assert.fail("Setup incomplete, tests will fail");
        }
```

This one confused me a bit in TestDeleteS3Object.testSimpleDelete. I doesn't quite match what the old testSimpleDelete method did, which was upload with key of hello.txt, with file hello.txt. I think you're uploading with key "folder" and with file "/hello.txt". I reasonably believe this does the same thing, but it was hard to read

```
    public void testSimpleDelete() throws IOException {
        // Prepares for this test
        uploadTestFile("folder");

        final TestRunner runner = TestRunners.newTestRunner(new DeleteS3Object());

        runner.setProperty(DeleteS3Object.CREDENTAILS_FILE, CREDENTIALS_FILE);
        runner.setProperty(DeleteS3Object.REGION, REGION);
        runner.setProperty(DeleteS3Object.BUCKET, BUCKET_NAME);
        runner.setProperty(DeleteS3Object.KEY, "folder");
```

On a similar note I think TestDeleteS3Object.testSimpleDeleteFolder was intended to delete with a key with a slash in it, as it was in the original test

```
    @Test
    public void testDeleteFolder() throws IOException {
        // Prepares for this test
        uploadTestFile("folder");
```

I think that you may have propagated a bug from TestDeleteS3Object.testTryToDeleteNotExistingFile [1]. I think the original test that second setting of BUCKET was erroneous. I think this should be BUCKET_NAME instead of "no-such-a-key".

```
        runner.setProperty(DeleteS3Object.BUCKET, "no-such-a-key");
```

Also, any thought to trying to exercise the Attribute Evaluation of the KEY in the tests? All the KEY properties were set explicitly. 

[1] https://github.com/apache/nifi/blob/master/nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/test/java/org/apache/nifi/processors/aws/s3/TestDeleteS3Object.java#L122

**Comment by trkurc** [comment]
Also, as I mentioned inline in the commit, this really isn't a unit test as much as an informational log. 
```
    @Test
    /**
     * Run this test to check what version of Joda-time is actually being linked.
     *
     * @see: https://issues.apache.org/jira/browse/NIFI-1025
     * @see: https://github.com/aws/aws-sdk-java/issues/444
     */
    public void testJodaTime() {
        System.out.println(new DateTime().getClass().getProtectionDomain().getCodeSource());
    }
```

**Comment by trkurc** [comment]
In TestFetchS3Object.testSimpleGet(), the key, "folder/" is a bit confusing. I'm not intensely familiar with the S3 java API, but I'm not sure how this fits with their "folder" concept [1]. 

```
    public void testSimpleGet() throws IOException {
        String key = "folder/";
        uploadTestFile(key);
```

Same issue with TestFetchS3Object.testTryToFetchNotExistingFile as TestDeleteS3Object.testTryToDeleteNotExistingFile. I think this tests for a non-existent bucket, not file.

[1] http://docs.aws.amazon.com/AmazonS3/latest/UG/FolderOperations.html

**Comment by trkurc** [comment]
@vs186031 - We're going to try to get this into 0.4.0 release of NiFi - I can make the changes, or if you have a patch in progress, I'll work off of that

**Comment by trkurc** [comment]
@vs186031 I've started making some of these changes

**Pull request closed without merging**